badge_server/test_badges.py

```diff
@@ -593,5 +626,358 @@ def test_github_py2py3_py3_incompatible_fresh_nodeps(self):
         self.assertEqual(json_response['left_text'],
                          'compatibility check (master)')
         self.assertEqual(json_response['right_text'], 'self incompatible')
+
+class TestBadgeImageOutdatedDependency(BadgeImageTestCase):
+
+    def test_pypi_py2py3_off_by_minor(self):
+        old_dep_info = dict(UP_TO_DATE_DEPS)
+        old_dep_info['google-auth'] = dict(UP_TO_DATE_DEPS['google-auth'])
```

Comment:
This is just a thought...maybe it makes sense to just add this single old dependency to the compatibility result rather than partially re-writing the existing dependency. It just saves have to look at the constant to see what changed. e.g.

```
old_dep = {
        'current_time': datetime.datetime.utcnow(),
        'installed_version': '1.4.0',
        'installed_version_time': datetime.datetime(
            2019, 2, 19, 21, 15, 56),
        'is_latest': False,
        'latest_version': '1.4.0',
        'latest_version_time': datetime.datetime(
            2019, 2, 19, 21, 15, 56)
}
old_dep_compat_results = list(RECENT_SUCCESS_DATA)
old_dep_compat_results.remove(GOOGLE_API_CORE_RECENT_SUCCESS_2)
old_dep_compat_results.remove(GOOGLE_API_CORE_RECENT_SUCCESS_3)
old_dep_compat_results.append(
            GOOGLE_API_CORE_RECENT_SUCCESS_2.with_updated_dependency_info(
                old_dep_info))
old_dep_compat_results.append(
            GOOGLE_API_CORE_RECENT_SUCCESS_3.with_updated_dependency_info(
                old_dep_info))
```

I think that the existing compatibility results in `RECENT_SUCCESS_DATA` should also include dependency info, at least for the package itself (because that is the actual behavior).

Reply:
I agree that rewriting the entire dependency instead of partially would be more clear - I will start updating the test cases.

With regards to `RECENT_SUCCESS_DATA` including dep info, I'm fine with that being implemented, but probably not in this PR.